### PR TITLE
[PoC] First attempt at MSVC setup - don't merge

### PR DESCRIPTION
**Pull request by mrikola**
@Interrupt asked for this, so here it is.

Prerequisites: Windows 10 SDK, MSVC 17, low-key masochism.

### CODE REVIEW

File: src/Libraries/RES/Source/res.h [hunk 1]
```diff
@@ -211,8 +215,10 @@ extern ResDesc2 *gResDesc2;
 
 extern Id resDescMax; // max id in res desc
 
+#define MAGICAL_ZERO_OFFSET 0x66E6E6E
+
```
Comment (mrikola) [inline src/Libraries/RES/Source/res.h]: This is basically voodoo.

**Comment by mrikola** [comment]
Notable things include:

**Changes I ended up making to appease the linker gods:**

src/Libraries/RES/Source/refacc.c, src/GameSrc/gamewrap.c: seemingly unnecessary <unistd.h>

src/GameSrc/setup.c: <unistd.h> for F_OK -> replace with numeric value of 0

src/Libraries/RES/Source/resbuild.c: on Win32 use _chsize from <io.h> over ftruncate from <unistd.h>

src/MacSrc/Modding.c: #include <dirent.h> is trouble, commented out whatever uses it

In src/Libraries/EDMS/Source/MODELS/: 

* d_f_2.cc, d_frame.cc, http://ftl.cc , http://pelvis.cc , http://robot.cc : surround #include "ss_flet.h" with extern "C" {}

* extern "C" magic also applied to:
  - d_f_face.cc: EDMS_Dirac_basis 
  - http://pelvis.cc : EDMS_pelvis_is_climbing

** That thing in `res.h` **

** Whatever is going on with the `Rect` initialization in `wrapper.c` **

** Filling empty structs with placeholder members **

**Comment by mrikola** [comment]
On a more serious note, I tried throwing the changed code at GCC, and it compiled and ran pretty much fine after I changed the voodoo constant back to 0 (which is the assumed value outside MSVC world) and added another `#include` somewhere. In other words, I haven't managed to find a place where I totally break everything (other than modding, which I commented out because of headers).

**Comment by Interrupt** [comment]
Thanks! This is super useful to look at, should be useful as a base for getting real MSVC support in here.

File: .gitignore [hunk 2]
```diff
@@ -48,6 +48,16 @@ TestSimpleMain
...
+# Visual Studio
+.vs
+Debug
+**/.vscode
+*.vcxproj*
+*.sln
+*.exp
+*.pdb
+*.sln
+
```
Comment (winterheart) [inline .gitignore]: Don't do that. Just mkdir build somewhere, cmake -G MSVC <sourcedir>, and play with compilation here.

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,15 +5,22 @@ cmake_minimum_required(VERSION 3.1)
 # Required for stdbool.h
 set(CMAKE_C_STANDARD 99)
 
-set(CMAKE_C_FLAGS -m32)
-set(CMAKE_CXX_FLAGS -m32)
+# This was useful when figuring out what's going wrong
+# set(CMAKE_VERBOSE_MAKEFILE true)
```
Comment (winterheart) [inline CMakeLists.txt]: Removed working code, fails on working systems

File: CMakeLists.txt [hunk 4]
```diff
@@ -5,15 +5,22 @@ cmake_minimum_required(VERSION 3.1)
...
-# HAAAAX!!
-add_definitions(-Dpowerc -DSVGA_SUPPORT)
+# TODO compile the main SDL and make a similar thing for it
+# right now my SDL installation is found via ${SDL2_DIR}
+set(VC_LIB_PATH_SUFFIX VisualC/Win32/Debug)
+set(CMAKE_FIND_LIBRARY_PREFIXES "")
+set(CMAKE_FIND_LIBRARY_SUFFIXES .lib)
+
+set(SDL2_PATH "build_ext/SDL2-2.0.8/")
+set(SDL2_MIXER_PATH "build_ext/SDL_mixer")
```
Comment (winterheart) [inline CMakeLists.txt]: Don't place SDL compilation things into main project, create separate or use precompiled.

File: CMakeLists.txt [hunk 5]
```diff
@@ -37,6 +44,10 @@ include_directories(
 message(STATUS "Found SDL2 Include:${SDL2_INCLUDE_DIRS} Library:${SDL2_LIBRARIES}")
 message(STATUS "Found SDL2_Mixer Include:${SDL2_MIXER_INCLUDE_DIRS} Library:${SDL2_MIXER_LIBRARIES}")
 
+# not actually sure if this does anything under MSVC...
+set(CMAKE_C_FLAGS "/I${SDL2_INCLUDE_DIRS}")
```
Comment (winterheart) [inline CMakeLists.txt]: Fails on Linux

**Comment by winterheart** [comment]
Well, I don't follow what this PR intended to do. It's MSVC support? Why don't use CMake -G MSVC generator?

**Comment by mrikola** [comment]
Yeah, sorry, I didn't communicate clearly enough that this isn't meant to be merged. I said over Twitter DM that I got this somewhat working, and Cuddigan requested a PR. Thanks for the suggestions, though. I'm totally new to CMake so any tips are appreciated.

File: .gitignore (hunk 2 above)
Comment (Interrupt) [reply .gitignore]: I'm okay with eventually having these in the gitignore, I can see us having a `generate-solution.bat` file eventually that runs the CMake command to make these files for people

**Comment by mrikola** [comment]
Closing to avoid confusion.

**Pull request closed without merging**